nifty-core/src/main/java/com/facebook/nifty/core/NiftyDispatcher.java

```diff
@@ -140,7 +140,8 @@ private void processRequest(
             // more requests that were in the latest read, even while further reads on the channel
             // have been blocked.
             if (requestSequenceId > lastResponseWrittenId.get() + queuedResponseLimit &&
-                !DispatcherContext.isChannelReadBlocked(ctx))
+                !DispatcherContext.isChannelReadBlocked(ctx) &&
+                DispatcherContext.isResponseOrderingRequired(ctx))
```

Comment:
Maybe do this check earlier because it is only needed for channels requiring order?

if (DispatcherContext.isResponseOrderingRequired(ctx)) {
  synchronized (responseMap)
  {
    if (...)
    {
          
    }
  }
}